php/class-media.php

```diff
@@ -1349,7 +1352,8 @@ private function create_attachment( $asset, $public_id ) {
 		update_post_meta( $attachment_id, '_' . md5( $sync_key ), true );
 		// record a base to ensure primary isn't deleted.
 		update_post_meta( $attachment_id, '_' . md5( 'base_' . $public_id ), true );
-
+		// capture the delivery type.
+		update_post_meta( $attachment_id, Sync::META_KEYS['delivery'], $asset['type'] );
```

Comment:
No. At this point, our internal doesn't exist yet.
After here, it goes to down sync, which then creates the attachment meta, based on the file that was downloaded.

Reply:
@pereirinha - Also, the internal getter is built to move standard metadata to the internal on the first internal request. This is by design, since we can't always update the internal. For example working with a v1 image, this does part of the v2 upgrade.

Reply:
@pereirinha - I just realised that the internal setter, actually uses the standard `update_post_meta` if the attachment meta doesn't exist. 😂 - Lets chat on this when you're online.